job/server/src/main/java/alluxio/worker/job/task/TaskExecutor.java

```diff
@@ -47,50 +49,66 @@
    *
    * @param jobId the job id
    * @param taskId the task id
-   * @param jobConfig the job configuration
-   * @param taskArgs the arguments passed to the task
+   * @param runTaskCommand the run task command
    * @param context the context on the worker
    * @param taskExecutorManager the task executor manager
    */
-  public TaskExecutor(long jobId, long taskId, JobConfig jobConfig, Serializable taskArgs,
+  public TaskExecutor(long jobId, long taskId, RunTaskCommand runTaskCommand,
```

Comment:
is handling task failure costly enough that we should do the basic checks before submitting to the task executor?

Reply:
The biggest cost is Deserialization. It's actually unclear to me that the previous version was even reporting the errors properly if deserialization failed.